pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -115,58 +115,74 @@ func removeDup(refs ResolvedResultRefs) ResolvedResultRefs {
 // then a nil list and error is returned instead.
 func convertToResultRefs(pipelineRunState PipelineRunState, target *ResolvedPipelineTask) (ResolvedResultRefs, string, error) {
 	var resolvedResultRefs ResolvedResultRefs
-	for _, ref := range v1.PipelineTaskResultRefs(target.PipelineTask) {
-		resolved, pt, err := resolveResultRef(pipelineRunState, ref)
-		if err != nil {
-			return nil, pt, err
+	for _, resultRef := range v1.PipelineTaskResultRefs(target.PipelineTask) {
+		referencedPipelineTask := pipelineRunState.ToMap()[resultRef.PipelineTask]
+		if referencedPipelineTask == nil {
+			return nil, resultRef.PipelineTask, fmt.Errorf("could not find task %q referenced by result", resultRef.PipelineTask)
+		}
+		if !referencedPipelineTask.isSuccessful() && !referencedPipelineTask.isFailure() {
+			return nil, resultRef.PipelineTask, fmt.Errorf("task %q referenced by result was not finished", referencedPipelineTask.PipelineTask.Name)
```

Comment:
could you help me understand this part? Under what cases we need to return errors. And which test case is testing this?

Reply:
@Yongxuanzhang This is just refactoring the existing code in `resolveResultRef()`
https://github.com/tektoncd/pipeline/blob/d26a45c048fa396f402332252a1cae641d2875fe/pkg/reconciler/pipelinerun/resources/resultrefresolution.go#L133C2-L135C3

Reply:
oh ok